backend/src/main/scala/bloop/util/HashedSource.scala

```diff
@@ -0,0 +1,35 @@
+package bloop.util
+
+import java.io.ByteArrayInputStream
+import java.io.InputStream
+import java.nio.file.Path
+
+import bloop.reporter.Problem
+
+import sbt.internal.inc.HashUtil
+import sbt.internal.inc.PlainVirtualFileConverter
+import xsbti.BasicVirtualFileRef
+import xsbti.FileConverter
+import xsbti.PathBasedFile
+import xsbti.VirtualFile
+
+case class HashedSource(content: Array[Byte], bloopHash: Int, path: Path)
```

Comment:
This is the biggest change, we use our own VirtualFile with eagerly read input so that we don't compile files with a hash that doesn't correspond to them

Reply:
I'd double check whether it should be a case class. BasicVirtualFileRef implements an equals method

Reply:
Changed to normal class